planning/behavior_path_planner/behavior_path_planner_path_generation.md

```diff
@@ -0,0 +1,91 @@
+## Path Generation
+
+This document explains how the path is ganerated for lane change and avoidance, etc.
+
+### Overview
+
+The base idea of the path generation in lane change and avoidance is to smoothly shift the reference path, such as the center line, in the lateral direction. This is achieved by using a constant-jerk profile as in the figure below. More details on how it is used can be found in [README](./README). It is assumed that the reference path is smooth enough for this algorithm.
+
+The figure below explains how the application of a constant lateral jerk $l^{'''}(s)$ can be used to induce lateral shifting. In order to comply with the limits on lateral acceleration and velocity, zero-jerk time is employed in the figure ( $T_a$ and $T_v$ ). In each interval where constant jerk is applied, the shift position $l(s)$ can be characterized by a third-degree polynomial. Therefore the shift length from the reference path can then be calculated by combining spline curves.
+
+<p align="center">
+  <img src="./image/path_shifter.png" width="800">
+</p>
+
+### Mathematical Derivation
+
+By applying simple integral operations, the following analytical equations can be derived to describe the shift distance $l(t)$ at each time under lateral jerk, acceleration, and velocity constraints.
+
+```math
+\begin{align}
+l_1&= \frac{1}{6}jT_j^3\\[10pt]
+l_2&= \frac{1}{6}j T_j^3 + \frac{1}{2} j T_a T_j^2 + \frac{1}{2} j T_a^2 T_j\\[10pt]
+l_3&= j  T_j^3 + \frac{3}{2} j T_a T_j^2 + \frac{1}{2} j T_a^2 T_j\\[10pt]
+l_4&= j T_j^3 + \frac{3}{2} j T_a T_j^2 + \frac{1}{2} j T_a^2 T_j + j(T_a + T_j)T_j T_v\\[10pt]
+l_5&= \frac{11}{6} j T_j^3 + \frac{5}{2} j T_a T_j^2 + \frac{1}{2} j T_a^2 T_j + j(T_a + T_j)T_j T_v \\[10pt]
+l_6&= \frac{11}{6} j T_j^3 + 3 j T_a T_j^2 + j T_a^2 T_j + j(T_a + T_j)T_j T_v\\[10pt]
+l_7&= 2 j T_j^3 + 3 j T_a T_j^2 + j T_a^2 T_j + j(T_a + T_j)T_j T_v
+\end{align}
+```
+
+These equations are used to determine the shape of a path. Additionally, by applying further mathematical operations to these basic equations, the following expressions can be derived.
```

Comment:
```suggestion
These equations are used to determine the shape of a path. Additionally, by applying further mathematical operations to these basic equations, the expressions of the following subsections can be derived.
```

Reply:
The user might find where is the expression, so I make the suggestion for better clarity. But if it is not necessary, then it's okay.

Reply:
Nice! Thank you!